### ISSUE TITLE
`noframenumbering` reduces the page number twice

### ISSUE THREAD

**Issue by ishiihidemasa** [elauksap/focus-beamertheme#29]
I recently started using `focus` theme. I like how it looks.
My trouble is that, when I use section page and focus frame, the page numbering fails.
To illustrate, I downloaded `focus-demo.tex`, `focus-demo_bibliography.bib`, and `focus-logo.pdf` and compiled the tex file on my PC.
I attach the generated PDF (that was converted to images).
I observe that the page number is reduced twice, though it should be reduced only once, every time a frame with `noframenumbering` option is created.

The version of `beamertheme-focus` is 3.3.0 (I believe it's the latest).
I use lualatex with the following `.latexmkrc` file.
Files are created in `build` directory.
```
$latex = 'lualatex -synctex=1 -interaction=nonstopmode -file-line-error -halt-on-error -outdir=build %O %S';
$out_dir = 'build';

# Biber, BibTeX
$biber = 'biber %O --bblencoding=utf8 -u -U --output_safechars %B';
$bibtex = 'upbibtex %O %B';
# makeindex
$makeindex = 'upmendex %O -o %D %S';

$pdf_mode = 4;  # lualatex

$max_repeat = 5;
```
I use Windows 11 and I usually work on VS Code with LaTeX Workshop extension.
I don't think the use of VS Code is relevant though, because the result doesn't change even if I run `latexmk focus-demo.tex` on terminal.

I'm not sure if it helps, but I tried several things by myself.
It seems that `\addtocounter{framenumber}{-1}` in `beamerouterthemefocus.sty` (line 210) is called twice.
I added another counter `mytest` and overrode the definition of `noframenumbering` as follows, so that `mytest` increases by one when `noframenumbering` frame is created.
```
\newcounter{mytest}
\defbeamertemplate{footline}{mynoframenumbering}{  % new attribute mynoframenumbering
    \addtocounter{framenumber}{-1}
    \addtocounter{mytest}{1}  % NEW
}
% Enable noframenumbering option.
\define@key{beamerframe}{noframenumbering}[true]{%
    \setbeamertemplate{footline}[mynoframenumbering]%  use mynoframenumbering instead
    \setlength{\focus@pbar@height}{0cm}%
    \focus@calculatefootheight%
}
```
It turned out that, when `mytest` was `0` before the focus frame (i.e. `mytest` is `0` on page $N$ and page $N + 1$ is a focus frame), `mytest` became `2` after the focus frame (i.e. on page $N + 2$).
I believe it should be `1` on page $N + 2$.

This is the first time I create an issue, and I apologize if I've done anything inappropriately.
Thank you in advance.

Hidemasa

![focus-demo_01](https://github.com/pcafrica/focus-beamertheme/assets/56109996/afcb9607-dc67-415b-aa53-091f11368e2d)
![focus-demo_02](https://github.com/pcafrica/focus-beamertheme/assets/56109996/7ec490db-1a87-435b-a9ba-bb017a0d24f8)
![focus-demo_03](https://github.com/pcafrica/focus-beamertheme/assets/56109996/558209b9-37bc-4f2d-85d4-5077760f8938)
![focus-demo_04](https://github.com/pcafrica/focus-beamertheme/assets/56109996/2bdd6e94-99bd-47a7-a416-a23e0d6a4f10)
![focus-demo_05](https://github.com/pcafrica/focus-beamertheme/assets/56109996/09ab78be-ba20-4971-82ac-6c9ab9d47c84)
![focus-demo_06](https://github.com/pcafrica/focus-beamertheme/assets/56109996/ed201a2e-587a-41df-815c-63df29cf2580)
![focus-demo_07](https://github.com/pcafrica/focus-beamertheme/assets/56109996/b122c253-f1fc-4c6c-9166-f5c4e576519e)
![focus-demo_08](https://github.com/pcafrica/focus-beamertheme/assets/56109996/ba5ec38b-2a79-4000-a09c-80b5a788af73)
![focus-demo_09](https://github.com/pcafrica/focus-beamertheme/assets/56109996/8df3d231-0d38-47da-b039-fabd3f180744)
![focus-demo_10](https://github.com/pcafrica/focus-beamertheme/assets/56109996/2b4c0f01-5d7b-44e0-b7f9-bcfa11cf3cb6)
![focus-demo_11](https://github.com/pcafrica/focus-beamertheme/assets/56109996/78a6c56e-3b6e-43dd-8ec4-8bdd0825302c)
![focus-demo_12](https://github.com/pcafrica/focus-beamertheme/assets/56109996/55e88ca8-6089-4a85-bd51-25bed5bbf5ea)
![focus-demo_13](https://github.com/pcafrica/focus-beamertheme/assets/56109996/08aa57c3-c720-4200-8321-7a4598281d5f)
![focus-demo_14](https://github.com/pcafrica/focus-beamertheme/assets/56109996/7d09988c-195e-4310-b1d2-4eeff18d8082)
![focus-demo_15](https://github.com/pcafrica/focus-beamertheme/assets/56109996/607b93ac-f55a-4f93-9a03-6cf30aed42f1)
![focus-demo_16](https://github.com/pcafrica/focus-beamertheme/assets/56109996/4bf540fd-cbec-428e-993d-db378b68da38)
![focus-demo_17](https://github.com/pcafrica/focus-beamertheme/assets/56109996/ea3a384b-5e8d-4a3a-865f-b2903ee07e9b)
![focus-demo_18](https://github.com/pcafrica/focus-beamertheme/assets/56109996/aaaaed65-7e74-4065-828b-72e1dd39ff30)

**Comment by pcafrica** [comment]
Hi @ishiihidemasa,

thanks for using Focus and for reporting the issue.

Unfortunately, I'm not able to reproduce the issue on any of my local machines. Even with your `.latexmkrc` file the generated PDF output looks correctly numbered.

Could you please provide a minimal working example? Also, which OS and LaTeX distribution are you using?

Best.

**Comment by ishiihidemasa** [comment]
Thank you very much for your prompt reply.
OS: Microsoft Windows 11 Home (10.0.22621)
LaTeX distribution:
The output says `This is LuaHBTeX, Version 1.17.0 (TeX Live 2023) `. (Does this answer your question?)

As for MWE, I came up with the following:
```
\documentclass{beamer}
\usetheme{focus}

\begin{document}
\begin{frame}[focus]
  This is focus frame!
\end{frame}

\begin{frame}{first slide}
  Page number should be one but it's zero.
\end{frame}
\end{document}
```
Compiling this tex file, I get the following PDF.
Page number should be one after the focus frame, but it is zero instead.

Hope it helps.
Thank you again for your response.

![mwe_1](https://github.com/pcafrica/focus-beamertheme/assets/56109996/34aa680b-822b-4470-b7a4-4aaf570e0d04)
![mwe_2](https://github.com/pcafrica/focus-beamertheme/assets/56109996/76fa6392-d8ab-4f4a-a618-9752529f6f27)

**Comment by pcafrica** [comment]
Unfortunately I'm still unable to reproduce the issue. Compiling your MWE with LuaLaTeX 1.14.0 outputs the following PDF, which is numbered correctly:
[texstudio_CqdbGz.pdf](https://github.com/pcafrica/focus-beamertheme/files/14195019/texstudio_CqdbGz.pdf)

There might be some incompatibility with the compiler version.

Have you tried to run the compiler more than once on the same file, or could you please try on some online platform like overleaf?

Best.

**Comment by pcafrica** [comment]
Update: it seems that this is caused by a different behaviour between TeXLive 2022 and TeXLive 2023...

**Comment by pcafrica** [comment]
I think I managed to fix it! Please find the fixed version in the latest release [v3.4.0](https://github.com/pcafrica/focus-beamertheme/releases/tag/v3.4.0).

Thanks again for helping to improve Focus!

Best.
